src/lib.rs

```diff
@@ -2,6 +2,7 @@
 
 #![forbid(unsafe_code)]
 #![deny(missing_docs)]
+#![recursion_limit = "256"]
```

Comment:
If this is only due to `quick-error`, I'd appreciate a comment here. It'd be slightly odd encountering this in the library otherwise.

Reply:
Done. I think it might only be needed because we're using `quick-error` on an error with several dozen enum variants